cluster_test.go

```diff
@@ -773,6 +773,43 @@ func TestClusterPins(t *testing.T) {
 	}
 }
 
+func TestClusterPinsWithExpiration(t *testing.T) {
+	ctx := context.Background()
+	cl, _, _, _ := testingCluster(t)
+	defer cleanState()
+	defer cl.Shutdown(ctx)
+
+	c := test.Cid1
+	opts := api.PinOptions{
+		ExpireAt: time.Now().Add(40 * time.Second),
+	}
+	_, err := cl.Pin(ctx, c, opts)
+	if err != nil {
+		t.Fatal("pin should have worked:", err)
+	}
+
+	pinDelay()
+
+	pins, err := cl.Pins(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(pins) != 1 {
+		t.Error("pin should be part of the state")
+	}
+
+	// try after expiry time and state sync interval
+	time.Sleep(cl.config.StateSyncInterval + 1*time.Second)
```

Comment:
The StateSyncInterval is set to 1 minute, so all the test suite is going to spend 59 seconds sleeping and last 10% more thanks to this. Reduce to 1 or 2 seconds (config_test.go, hopefully doesn't break something else).

Reply:
Instead of changing state sync interval for all tests, I have changed only for pin expiration test.

This would avoid statesync running too often in tests. 

Just having a global variable to do this is much less code change then being able to pass configs while creating cluster peers.